installer/bundle/bundle_skel_Linux.sh

```diff
@@ -513,10 +521,13 @@ then
 
     if [ "$installMode" = "P" ]
     then
-        echo "Purging all files in cross-platform agent ..."
-        rm -rf /etc/opt/microsoft/*-cimprov /etc/opt/microsoft/scx /opt/microsoft/*-cimprov /opt/microsoft/scx /var/opt/microsoft/*-cimprov /var/opt/microsoft/scx
-        rmdir /etc/opt/microsoft /opt/microsoft /var/opt/microsoft 1>/dev/null 2>/dev/null
-
+        # Remove directoried only if scx got removed successfully (Other products might be dependent on scx)
+        check_if_pkg_is_installed scx
+        if [ $? -ne 0 ]; then
+            echo "Purging all files in cross-platform agent ..."
+            rm -rf /etc/opt/microsoft/*-cimprov /etc/opt/microsoft/scx /opt/microsoft/*-cimprov /opt/microsoft/scx /var/opt/microsoft/*-cimprov /var/opt/microsoft/scx
```

Comment:
You shouldn't be nuking the *-cimprov directories unless underlying packages are removed.

So don't nuke .../apache-cimprov unless the apache package is removed. Same for MySQL.